### PR TITLE
fix warning from using deprecated TSC URL API

### PR DESCRIPTION
**Pull request by tomerd**
motivation: TSC URL is deprecated

changes: migrate the url scheme extraction code to the one call site that needs it

### CODE REVIEW

File: Sources/PackageLoading/ManifestJSONParser.swift [hunk 1]
```diff
@@ -415,6 +411,33 @@ enum ManifestJSONParser {
         )
     }
 
+    /// Parses the URL type of a git repository
+    /// e.g. https://github.com/apple/swift returns "https"
+    /// e.g. git@github.com:apple/swift returns "git"
+    ///
+    /// This is *not* a generic URI scheme parser!
+    private static func parseScheme(_ location: String) -> String? {
```
Comment (tomerd) [inline Sources/PackageLoading/ManifestJSONParser.swift:419]: we can potentially also use Foundation API here, although not obvious which. for now migrated the code as is to preserve backwards compatibility. open to other ideas.

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/PackageLoading/ManifestJSONParser.swift (hunk 1 above)
Comment (abertelrud) [reply Sources/PackageLoading/ManifestJSONParser.swift:419]: Probably Foundation.URLComponents could help.  It does fall down pretty badly on the second one though.  So maybe try URLComponents(...).scheme first and then special-case `git@` prefix if that fails?  Not sure how much that actually buys.

Comment (tomerd) [reply Sources/PackageLoading/ManifestJSONParser.swift:419]: tried URLComponents but it does not handle the git@ prefix. @millenomi ?

**Review by neonichu — APPROVED** [review]

**Pull request merged**